sphinxcontrib/needs/filter_common.py

```diff
@@ -209,6 +218,11 @@ def filter_needs(needs, filter_string="", current_need=None):
 
     found_needs = []
 
+    # needs external filter data
+    if NEEDS_EXTERNAL_FILTER_DATA:
+        for extern_filter, value in NEEDS_EXTERNAL_FILTER_DATA.items():
+            filter_string = filter_string.replace(extern_filter, '"{}"'.format(value))
```

Comment:
That kind of handling is not needed.

Just add NEEDS_EXTERNAL_FILTER_DATA to the context of the eval()-func

Reply:
Done.